### PR TITLE
Fixes #3273. Updates `FindDeepestView` to support `Adornment`

### LINKED ISSUE #3273: Update `FindDeepestView` to support `Adornment`
- There are no dedicated unit tests for `FindDeepestView` right now. We need some. Write some that test the current `v2_develop` version that prove it works.

- The current version ignores `Adornments` and returns a view if x/y are anywhere within `view.Frame`. 

- Simplify `FindDeepestView` to NOT have the `out` params. 
  
- Modify `FindDeepestView` to return `Margin`, `Padding`, or `Border` if `x`/`y` match. In `Application.OnMouseEvent`, temporarily ignore these. Eg. like this:

```cs
        var view = View.FindDeepestView (Current, a.MouseEvent.X, a.MouseEvent.Y, out int screenX, out int screenY);

        // TODO: Remove this temporary filter:
        if (view is Adornment adornment)
        {
            view = adornment.Parent;
        }
```

- Add `FindDeepestView` unit tests to prove this change works.

- Delete all the `Adornment`-specific code in `Application.OnMouseEvent` and add `MouseEvent` handling code to `Adornment`, `Margin`, `Border`, and `Padding` as appropriate (e.g. `Adornment.MouseClick += Parent.SetFocus()`).

- Don't worry about `Adornment.SubViews` yet. That should happen in either THIS PR or the `Close Box` PR.

### PR DESCRIPTION
**Pull request by tig**
## Fixes

- Fixes #3273 
- See https://github.com/gui-cs/Terminal.Gui/pull/3254#discussion_r1503486273

## Proposed Changes/Todos

- [x] Write some that test the current `v2_develop` version that prove it works.
- [x] Simplify `FindDeepestView` to NOT have the `out` params. Modify callers to call `view.FrameToScreen` on the returned view instead. 
- [x] Fix bugs mentioned here: https://github.com/gui-cs/Terminal.Gui/pull/3254#discussion_r1503486273
- [ ] The current version ignores `Adornments` and returns a view if x/y are anywhere within `view.Frame`.  Modify `FindDeepestView` to return `Margin`, `Padding`, or `Border` if `x`/`y` match. In `Application.OnMouseEvent`, temporarily ignore these. Eg. like this:

```cs
        var view = View.FindDeepestView (Current, a.MouseEvent.X, a.MouseEvent.Y, out int screenX, out int screenY);

        // TODO: Remove this temporary filter:
        if (view is Adornment adornment)
        {
            view = adornment.Parent;
        }
```

- [ ] Add more `FindDeepestView` unit tests to prove this change works.
- [ ] Delete all the `Adornment`-specific code in `Application.OnMouseEvent` and add `MouseEvent` handling code to `Adornment`, `Margin`, `Border`, and `Padding` as appropriate (e.g. `Adornment.MouseClick += Parent.SetFocus()`).

- Don't worry about `Adornment.SubViews` yet. That should happen later.

## Pull Request checklist:

- [x] I've named my PR in the form of "Fixes #issue. Terse description."
- [x] My code follows the [style guidelines of Terminal.Gui](https://github.com/gui-cs/Terminal.Gui/blob/develop/.editorconfig) - if you use Visual Studio, hit `CTRL-K-D` to automatically reformat your files before committing.
- [x] My code follows the [Terminal.Gui library design guidelines](https://github.com/gui-cs/Terminal.Gui/blob/develop/CONTRIBUTING.md)
- [x] I ran `dotnet test` before commit
- [x] I have made corresponding changes to the API documentation (using `///` style comments)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any poor grammar or misspellings
- [x] I conducted basic QA to assure all features are working

### CODE REVIEW

**Comment by tig** [comment]
@BDisp feel free to submit PRs to this. I have about an hour of time left today...

**Comment by tig** [comment]
@BDisp This now fixes the bugs I discussed here:

https://github.com/gui-cs/Terminal.Gui/pull/3254#discussion_r1503486273

File: Terminal.Gui/Application.cs [hunk 1]
```diff
@@ -1469,11 +1478,11 @@ internal static void OnMouseEvent (MouseEventEventArgs a)
 
             var me = new MouseEvent
             {
-                X = screenX,
-                Y = screenY,
+                X = a.MouseEvent.X - screen.X,
```
Comment (dodexahedron) [inline Terminal.Gui/Application.cs:1481]: At some point (incoming pun alert), it would be nice for any type we have that has a location to keep it as a Point struct, for consistency (or a Rectangle, if that's more relevant for the given type). X and Y can still be exposed for convenience, of course, and just proxy to the properties on the Point.

Using Point or Rectangle, as appropriate for each type, also gives nice stuff like the Offset method and others which can make quite a few spots in our code a lot cleaner, clearer, and more consistent, too. Plus it helps eliminate the work of having to debug that math for every new method we add.

**Review by dodexahedron — APPROVED** [review]
I don't see anything major here needing immediate attention.

Annotating or handling appropriately to address the null warnings would be nice but I think isn't worth it here because that code is already going to get touched again for event refactoring.

TL;DR: LGTM

**Comment by tig** [comment]
> I don't see anything major here needing immediate attention.
> 
> Annotating or handling appropriately to address the null warnings would be nice but I think isn't worth it here because that code is already going to get touched again for event refactoring.
> 
> TL;DR: LGTM

I'm going to merge the work done here so far as https://github.com/gui-cs/Terminal.Gui/pull/3277

File: Terminal.Gui/Application.cs (hunk 1 above)
Comment (tig) [reply Terminal.Gui/Application.cs:1481]: Agree!
